core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -64,9 +64,12 @@ protected void beforeComplete(Subscriber<? super U> subscriber) {}
     /**
      * A callback executed just before calling {@link Subscriber#onError(Throwable)} on {@code subscriber}.
      * Override this method to execute any cleanup logic that may be needed before failing the
-     * subscription.
+     * subscription. This method may rewrite the {@code cause} and then return a new one so that the new
+     * {@link Throwable} would be passed to {@link Subscriber#onError(Throwable)}.
      */
-    protected void beforeError(Subscriber<? super U> subscriber, Throwable cause) {}
+    protected Throwable beforeError(Subscriber<? super U> subscriber, Throwable cause) {
```

Comment:
Add `@Nullable` to the method.

Reply:
I think we should apply the same logic to `beforeSubscribe` as well.